src/Sarif/Schemata/sarif-schema.json

```diff
@@ -1419,9 +1417,7 @@
           "description": "Key/value pairs that provide additional information about the physical location.",
           "$ref": "#/definitions/propertyBag"
         }
-      },
-
-      "required": [ "artifactLocation" ]
```

Comment:
>"required": [ "artifactLocation" ] [](start = 6, length = 34)

A controversial change! Our spec clearly states that physicalLocation.artifactLocation is required! Then, why did i remove it? because of stackframe.location.PhysicalLocation - i was not able to figure out how to construct a new ArtifactLocation object for stackFrame. Maybe there's an answer to this, but... 

i dont think we can fulfill this criteria when we are moving legacy address info to stackframe.location.PhysicalLocation.Address.

thoughts? :)

Reply:
This s/be anyOf "address" or "artifactLocation" as required. Open a tracking item in the SDK repo.

---
In reply to: [270611226](https://github.com/Microsoft/sarif-sdk/pull/1387#discussion_r270611226) [](ancestors = 270611226)